src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/Gambatte.IEmulator.cs

```diff
@@ -9,56 +9,94 @@ public partial class Gameboy : IEmulator, IBoardInfo
 	{
 		public IEmulatorServiceProvider ServiceProvider { get; }
 
-		public ControllerDefinition ControllerDefinition => GbController;
+		public ControllerDefinition ControllerDefinition => ((int)_syncSettings.FrameLength == 2) ? SubGbController : GbController;
 
 		public bool FrameAdvance(IController controller, bool render, bool rendersound)
 		{
 			FrameAdvancePrep(controller);
-			if (_syncSettings.EqualLengthFrames)
+			uint samplesEmitted;
+			switch ((int)_syncSettings.FrameLength)

```

Comment:
Don't need to cast to `int` for equality or for `switch`.

Reply:
https://prnt.sc/135ohdh
https://prnt.sc/135oit9